### ISSUE TITLE
Determinant use * instead of self.mul 

### ISSUE THREAD

**Issue by GerfautGE** [emin63/pyfinite#10]
in genericmatrix.py line 601,  I read : "det = det * workingCopy.data[i][I]"
Should not it be :  "det = self.mul(det,  workingCopy.data[i][I])" instead ?

I observe this issue working with Genericmatrix in a finite field and determinant returns elements not include in the field ...

**Comment by emin63** [comment]
If the `*` operator is defined properly, shouldn't those give the same result?

In general when you are reporting a bug, it would be helpful to give an example of expected behavior vs actual behavior. For example, do you have an example where the current code produces an incorrect or unexpected answer? If so, that would be helpful in understanding the problem you are reporting, fixing it, and putting in an automated test for the future.

Thanks.

**Comment by GerfautGE** [comment]
```python
>>> import pyfinite.ffield as pffield
>>> import pyfinite.genericmatrix as pfmat
>>> F = pffield.FField(4)  #F16 
>>> G = pfmat.GenericMatrix((2, 2), zeroElement=0, identityElement=1, add=F.Add, sub=F.Add, mul=F.Multiply, div=F.Divide)
>>> G.SetRow(0, [15, 0])
>>> G.SetRow(1, [0, 15])
>>> G.Determinant()  # Actual behaviour
225
>>> F.Multiply(15, 15)  # Expected behaviour
10
```

As a polynomial regarding matrix's coefficients , the determinant must be in F16. But 225 does not represent an element of F16 in pyfinite.
I think replacing "det = det * workingCopy.data[i][I]" by "det = self.mul(det, workingCopy.data[i][I])" will fix the issue.

**Comment by emin63** [comment]
Thanks for that example. I see what you mean now.

I fixed the bug in PR #11 and pushed a new release as 1.8.

You can see a test case verifying your example in the new [test_basics.py](https://github.com/emin63/pyfinite/blob/master/pyfinite/test_basics.py) file.
